app/src/tables.js

```diff
@@ -52,6 +52,13 @@ function uncheckAll(vm) {
         vm.checkAllObs(false);
     };
 }
+function redirectHandler(vm, redirect) {
+    return function(request) {
+        if (vm.itemsObs().length === 0 && !!redirect) {
+            document.location = redirect;
+        }
+    };
```

Comment:
Hard to believe this hasn't been addressed before, and list/table failure handlers appear redundant.